pkg/install/deploystorage.go

```diff
@@ -91,41 +92,9 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 		}
 	}
 
-	var clusterSPObjectID string
-	{
-		spp := &i.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
-
-		token, err := aad.GetToken(ctx, i.log, i.doc.OpenShiftCluster, azure.PublicCloud.GraphEndpoint)
-		if err != nil {
-			return err
-		}
-
-		spGraphAuthorizer := autorest.NewBearerAuthorizer(token)
-
-		applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
-
-		timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
-		defer cancel()
-		// NOTE: Do not override err with the error returned by wait.PollImmediateUntil.
-		// Doing this will not propagate the latest error to the user in case when wait exceeds the timeout
-		wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-			var res azgraphrbac.ServicePrincipalObjectResult
-			res, err = applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
-			if err != nil {
-				if strings.Contains(err.Error(), "Authorization_IdentityNotFound") {
-					i.log.Info(err)
-					return false, nil
-				}
-
-				return false, err
-			}
-
-			clusterSPObjectID = *res.Value
-			return true, nil
-		}, timeoutCtx.Done())
-		if err != nil {
-			return err
-		}
```

Comment:
moved to a separate function, `clusterSPObjectID`, since it's now needed for both `deployStorageTemplate` and `deploySnapshotUpgradeTemplate`

(I would suggest opening up two windows for this PR to compare the moved code)